<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>800</width>
    <height>600</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>MainWindow</string>
  </property>
  <widget class="QWidget" name="centralwidget">
   <layout class="QHBoxLayout" name="horizontalLayout" stretch="1,3">
    <item>
     <widget class="QGroupBox" name="inputGroup">
      <property name="title">
       <string>输入</string>
      </property>
      <layout class="QVBoxLayout" name="verticalLayout" stretch="0,0,9,0,6">
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_4">
         <item>
          <widget class="QPushButton" name="chooseFile">
           <property name="text">
            <string>导入文件</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="saveFile">
           <property name="text">
            <string>保存文件</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_2">
         <item>
          <widget class="QLabel" name="label">
           <property name="text">
            <string>文法输入</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="toParseGrammer">
           <property name="text">
            <string>解析文法</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextEdit" name="grammer"/>
       </item>
       <item>
        <layout class="QHBoxLayout" name="horizontalLayout_3">
         <item>
          <widget class="QLabel" name="label_2">
           <property name="text">
            <string>分析LEX</string>
           </property>
          </widget>
         </item>
         <item>
          <widget class="QPushButton" name="toParseStatement">
           <property name="text">
            <string>分析</string>
           </property>
          </widget>
         </item>
        </layout>
       </item>
       <item>
        <widget class="QTextEdit" name="statement"/>
       </item>
      </layout>
     </widget>
    </item>
    <item>
     <widget class="QTabWidget" name="resultTab">
      <property name="currentIndex">
       <number>0</number>
      </property>
      <widget class="QWidget" name="result">
       <attribute name="title">
        <string>文法解析结果</string>
       </attribute>
       <layout class="QHBoxLayout" name="horizontalLayout_5" stretch="1,2">
        <item>
         <widget class="QGroupBox" name="outputBox">
          <property name="title">
           <string>文法解析基本信息</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_2" stretch="0,1,1,3,1,3,1,3">
           <item>
            <widget class="QLabel" name="label_7">
             <property name="text">
              <string>文法错误</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextBrowser" name="syntaxError"/>
           </item>
           <item>
            <widget class="QLabel" name="label_3">
             <property name="text">
              <string>拓广文法</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextBrowser" name="extraGrammer"/>
           </item>
           <item>
            <widget class="QLabel" name="label_4">
             <property name="text">
              <string>非终结符号First集合</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextBrowser" name="firstSet"/>
           </item>
           <item>
            <widget class="QLabel" name="label_5">
             <property name="text">
              <string>非终结符号Follow集合</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTextBrowser" name="followSet"/>
           </item>
          </layout>
         </widget>
        </item>
        <item>
         <widget class="QGroupBox" name="groupBox">
          <property name="title">
           <string>文法解析详情</string>
          </property>
          <layout class="QVBoxLayout" name="verticalLayout_3">
           <item>
            <widget class="QLabel" name="label_8">
             <property name="text">
              <string>LR(0) DFA</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTableWidget" name="dfa">
             <property name="sizePolicy">
              <sizepolicy hsizetype="Expanding" vsizetype="Expanding">
               <horstretch>0</horstretch>
               <verstretch>0</verstretch>
              </sizepolicy>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QLabel" name="label_9">
             <property name="text">
              <string>SLR(1) 分析表</string>
             </property>
            </widget>
           </item>
           <item>
            <widget class="QTableWidget" name="slr"/>
           </item>
          </layout>
         </widget>
        </item>
       </layout>
      </widget>
      <widget class="QWidget" name="syntaxTree">
       <attribute name="title">
        <string>语法树</string>
       </attribute>
       <layout class="QVBoxLayout" name="verticalLayout_4">
        <item>
         <widget class="QTreeWidget" name="treeWidget">
          <column>
           <property name="text">
            <string notr="true">1</string>
           </property>
          </column>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
   </layout>
  </widget>
 </widget>
 <resources/>
 <connections/>
</ui>
